adodb.inc.php

```diff
@@ -4064,7 +4067,10 @@ function __construct($queryID,$mode=false) {
 	}
 
 	function __destruct() {
-		$this->Close();
+		//If the _queryID is bogus, calling Close() may cause an error in the database driver.
+		if($this->_queryID != -1) {
+			$this->Close();
+		}
```

Comment:
Wouldn't it be more appropriate to add this fix in ADORecordSet::Close() ?

Reply:
Agreed, that would make more sense.

Reply:
Our internal ADODB branch has been out-of-sync with upstream for over a decade now. It's possible we missed some upstream changes that resolved some of our issues in other ways. 

These are some of the last bits to bring it back in sync though, so that will be nice. We can leave this out for now and we will run further tests and let you know if there still a problem.

Thanks for your prompt response with these PR's.

Reply:
You're welcome. So should I close this PR then ?

Reply:
Sure